app/modules/collaborations/models.py

```diff
@@ -17,13 +51,126 @@ class Collaboration(db.Model, HoustonModel):
+                )
+
+            collab_user_assoc = CollaborationUserAssociations(
+                collaboration=self, user=user_tup[0]
+            )
+
+            if (
+                approval_states is not None
+                and approval_states[i] in CollaborationUserState.ALLOWED_STATES
+            ):
+                collab_user_assoc.read_approval_state = approval_states[i]
+            if is_initiator is not None and isinstance(is_initiator[i], bool):
+                collab_user_assoc.initiator = is_initiator[i]
+                # If you initiate you approve
+                collab_user_assoc.read_approval_state = CollaborationUserState.APPROVED
+
+            with db.session.begin():
+                db.session.add(collab_user_assoc)
+                db.session.merge(user_tup[0])
+
+    def get_users(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            users.append(association.user)
+        return users
+
+    def get_initiators(self):
+        users = []
+        for association in self.collaboration_user_associations:
+            if association.initiator is True:
+                users.append(association.user)
+        return users
+
+    def get_read_state(self):
+        read_state = CollaborationUserState.APPROVED
+        for association in self.collaboration_user_associations:
+            if association.read_approval_state is CollaborationUserState.DECLINED:
+                read_state = CollaborationUserState.DECLINED
+            elif association.read_approval_state is CollaborationUserState.PENDING:
+                read_state = CollaborationUserState.PENDING
+        return read_state
+
+    def get_edit_state(self):
+        edit_state = CollaborationUserState.APPROVED
+        for association in self.collaboration_user_associations:
+            if association.edit_approval_state is CollaborationUserState.DECLINED:
+                edit_state = CollaborationUserState.DECLINED
+            elif association.edit_approval_state is CollaborationUserState.PENDING:
+                edit_state = CollaborationUserState.PENDING
+            elif association.edit_approval_state is CollaborationUserState.NOT_INITIATED:
+                edit_state = CollaborationUserState.NOT_INITIATED
+        return edit_state
+
+    def set_read_approval_state_for_user(self, user_guid, state):
+        from app.modules.users.models import User
+
+        user = User.ensure_edm_obj(user_guid)
+        if user is not None and state in CollaborationUserState.ALLOWED_STATES:
+            for association in self.collaboration_user_associations:
+                if association.user is user:
+                    association.read_approval_state = state
+
+    def set_edit_approval_state_for_user(self, user_guid, state):
+        from app.modules.users.models import User
+
+        user = User.ensure_edm_obj(user_guid)
+        if (
```

Comment:
This has no check for None user.
Thinking this might be clearer as one for loop with specific handling rather than two separate for loops

Reply:
I'll get a None check in and try to slim this down. I remember some case forcing me to use two loops but maybe with fresh eyes on it I can cut one out.